src/Symfony/Component/HttpKernel/Tests/Controller/ArgumentResolverTest.php

```diff
@@ -0,0 +1,236 @@
+        $request->attributes->set('foobar', 'foobar');
+        $controller = array(new self(), 'controllerWithFooBarFoobar');
+
+        try {
+            self::$resolver->getArguments($request, $controller);
+            $this->fail('->getArguments() throws a \RuntimeException exception if it cannot determine the argument value');
+        } catch (\Exception $e) {
+            $this->assertInstanceOf('\RuntimeException', $e, '->getArguments() throws a \RuntimeException exception if it cannot determine the argument value');
+        }
+    }
+
+    public function testGetArgumentsInjectsRequest()
+    {
+        $request = Request::create('/');
+        $controller = array(new self(), 'controllerWithRequest');
+
+        $this->assertEquals(array($request), self::$resolver->getArguments($request, $controller), '->getArguments() injects the request');
+    }
+
+    public function testGetArgumentsInjectsExtendingRequest()
+    {
+        $request = ExtendingRequest::create('/');
+        $controller = array(new self(), 'controllerWithExtendingRequest');
+
+        $this->assertEquals(array($request), self::$resolver->getArguments($request, $controller), '->getArguments() injects the request when extended');
+    }
+
+    /**
+     * @requires PHP 5.6
+     */
+    public function testGetVariadicArguments()
+    {
+        $request = Request::create('/');
+        $request->attributes->set('foo', 'foo');
+        $request->attributes->set('bar', array('foo', 'bar'));
+        $controller = array(new VariadicController(), 'action');
+
+        $this->assertEquals(array('foo', 'foo', 'bar'), self::$resolver->getArguments($request, $controller));
+    }
+
+    /**
+     * @requires PHP 5.6
+     * @expectedException \InvalidArgumentException
+     */
+    public function testGetVariadicArgumentsWithoutArrayInRequest()
+    {
+        $request = Request::create('/');
+        $request->attributes->set('foo', 'foo');
+        $request->attributes->set('bar', 'foo');
+        $controller = array(new VariadicController(), 'action');
+
+        self::$resolver->getArguments($request, $controller);
+    }
+
+    /**
+     * @requires PHP 5.6
+     * @expectedException \InvalidArgumentException
+     */
+    public function testGetArgumentWithoutArray()
+    {
+        $factory = new ArgumentMetadataFactory();
+        $valueResolver = $this->getMock(ArgumentValueResolverInterface::class);
+        $resolver = new ArgumentResolver($factory, array($valueResolver));
+
+        $valueResolver->expects($this->any())->method('supports')->willReturn(true);
+        $valueResolver->expects($this->any())->method('resolve')->willReturn('foo');
+
+        $request = Request::create('/');
+        $request->attributes->set('foo', 'foo');
+        $request->attributes->set('bar', 'foo');
+        $controller = array($this, 'controllerWithFooAndDefaultBar');
+        $resolver->getArguments($request, $controller);
+    }
```

Comment:
What about a test with an expected exception when `getDefaultValue()` is called on an argument which hasn't one?